Lib/urllib/request.py

```diff
@@ -1494,6 +1494,8 @@ def _is_local_authority(authority):
         if authority == hostname:
             return True
     # Compare IP addresses
+    if not resolve:
+        return False
     try:
         address = socket.gethostbyname(authority)
     except (socket.gaierror, AttributeError):
```

Comment:
```suggestion
    except (socket.gaierror, AttributeError, UnicodeEncodeError):
```

gethostbyname tries to encode with IDNA, which can fail (e.g. on the unencodable characters we put into paths for our test suite)